open-company/open-company-web
# Post sorting

Review after #910 

Part of https://trello.com/c/CJahkpWQ

Review with:
- [Storage](https://github.com/open-company/open-company-storage/pull/208)
- [Change](https://github.com/open-company/open-company-change/pull/21)
- [Lib](https://github.com/open-company/open-company-lib/pull/66)
- [Lib 2](https://github.com/open-company/open-company-lib/pull/67)
- [Infra](https://github.com/belucid/open-company-infrastructure/pull/109)

To test:
- boards pagination
- boards sort
- AP pagination
- AP sort
- NEW comment(s) tag
- drafts board (no sort, show always all the entries, no attempt to load more)
- sort dropdown style
- sort dropdown on mobile
- check 404 for not existing post uuid
- check current sort type is remembered and reused across sections and AP
- check live comments coming in from another user
- check section name update doesn't break pagination or section data: try switching sort after a board edit
- check pagination doesn't break when switching sort order: load more pages then switch sort then switch back
- check post move still work as expected
- check post delete still work as expected

Possible scenarios inherited from board pagination PR to fix and then test:
- direct load of a post within the first page
- direct load of a post out of the first page
- internal nav to a post within the first page
- internal nav to a post out of the first page (with a notification click for example)
- make sure pagination is not trying load the last page over and over but only one time
